### ISSUE TITLE
Azure DevOps Server - Deployment of a release with a (latest version) of an artifact (feed) is failing after the latest version was deleted

### ISSUE THREAD

**Issue by silviulucacentriceu** [microsoft/azure-pipelines-tasks#13689]
## Note  
I am unable to deploy a release that is using the latest version of a feed artifact, after the latest version of the artifact was deleted.
 
On AzureDevOps server, Version Dev17.M153.5.
One release that is using an artifact from a feed (latest version). 
The latest version of the artifact is deleted from feed (and from recycle been).
Create the release (default settings) - is using the deleted version of the artifact instead of the latest available one.
Deploy the release: error in download artifacts task - "... Not found:.

The only solution for this is to trigger a new build to create a new package with a newer version. This is not feasible, because builds, releases and deploys are handled by different teams.  

## Required Information

Entering this information will route you directly to the right team and expedite traction.

**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: Download Artifacts

list here (V# not needed):  
https://github.com/Microsoft/azure-pipelines-tasks/tree/master/Tasks

## Environment
- Server - Azure Pipelines or TFS on-premises?
    
    AzureDevOps server, Version Dev17.M153.5

- Agent - Hosted or Private: 
    
    Hosted Agent, Windows (nano, 2016)

## Issue Description

I am unable to deploy a release that is using the latest version of a feed artifact, after the latest version of the artifact was deleted.

### Task logs

##[error]Failed to download package file from XxxFeed got the following error: Error: 404: Not Found

## Troubleshooting
When creating the release, the version of the deleted artifact is selected instead of the version of the last available one.

### Error logs

##[error]Failed to download package file from XxxFeed got the following error: Error: 404: Not Found

**Comment by sdobrodeev** [comment]
Hi @silviulucacentriceu Could you please clarify which task you are using: DownloadBuildArtifacts@0, DownloadPipelineArtifact@2 or maybe something else?

**Comment by silviulucacentriceu** [comment]
Hi @sdobrodeev,
It is happening in the classic releases, when using a deployment group job with "Skip download of artifacts" on false (on AzureDevOps server, Version Dev17.M153.5, not sure if it is fixed on a newer version).
If this is not the right place for this thread, can you move the issue please to the right one?

**Comment by EzzhevNikita** [comment]
Hey @silviulucacentriceu , if I understand your problem in right way the behaviour that you described is expected. According to the [release artifacts documentation](https://docs.microsoft.com/en-us/azure/devops/pipelines/release/artifacts?view=azure-devops#:~:text=After%20a%20release%20is%20created,same%20artifacts%20in%20all%20stages.), the version of the artifacts could be specified only on the step of release creation. The latest version here means that it is the latest version on the moment of release creation not deploying. It is not possible to change this version after release was created, even if the specified version of the artifact has been removed.

The only workaround that I could suggest for this case is to create a new release manually and specify desired version of artifact while creation.

**Comment by silviulucacentriceu** [comment]
Hi @EzzhevNikita,

Thanks for the response.
The issue is still present: if I'm trying to create a new release (not to deploy it on some specific environments), the deleted version of the artifact is still used and I need to manually change it to one that exists: In "Create a new release" section, the "Version" dropdown includes also the deleted versions of the artifact (it should ignore the deleted ones).
I am not expecting to have the version updated during deployment, but I was expecting for the release to use the latest not deleted package. 

From some basic tests in Azure DevOps Server Version Dev18.M170.6, it seems that the issue is not reproducible anymore.

**Comment by EzzhevNikita** [comment]
Hi @silviulucacentriceu,

Sorry it's my bad, seems I misunderstood your problem at first. I've checked the case you have described and it is not reproducible on the latest version Azure DevOps. Seems the issue was fixed. So could you please confirm that there no this issue on the Dev18.M170.6 version?

**Comment by silviulucacentriceu** [comment]
Hi @EzzhevNikita,

I am not able yet to fully test it, because the test server was only updated with azure update: Dev18.M170.6 version. The database wasn't updated and because of this, I don't have all the releases from the production server in order to be able to look deeper into it (delete artifact and create a new release for one of our release definition). When I tested it on the test server, I just used a basic release with some test artifacts.

Once we will fully migrate the production server to the newer version, I will be able also to test the issue using the same release/artifacts as I used on the older version. If it will still be reproducible, I will add add a comment here (reopen the thread?).
For the moment at least, we can close it.
